File: 백준 문제 풀이 모음/python/class 3/11724.py
import sys
sys.setrecursionlimit(10000)

n, m = map(int, sys.stdin.readline().split())

adj = []
count = 0
visited= [False] * (n+1)
for i in range(n+1) :
    adj.append([])

def dfs(v) :
    visited[v] = True
    for e in adj[v] :
        if not visited[e] :
            dfs(e)

for i in range(m) :
    x, y = map(int, sys.stdin.readline().split())
    adj[x].append(y)
    adj[y].append(x)


for i in range(1, n+1) :
    if not visited[i] :
        dfs(i)
        count +=1


print(count)

File: 백준 문제 풀이 모음/kotlin/class 3/11724.kt

var matrix = MutableList<MutableList<Int>>(0) { MutableList(0) { 0 } }
var visited = MutableList<Int>(0) { 0 }

fun main() = with(System.`in`.bufferedReader()){
    val str = readLine().split(' ').toList()
    val n = str[0].toInt()
    val m = str[1].toInt()

    matrix = MutableList<MutableList<Int>>(n + 1) { MutableList(0) { 0 } }
    visited = MutableList<Int>(n + 1) { 1 }

    repeat(m) {
        val input = readLine().split(' ').toList()
        val a = input[0].toInt()
        val b = input[1].toInt()

        matrix[a].add(b)
        matrix[b].add(a)

    }

    var count = 0
    for (i in 1 .. n) {
       if (visited[i] == 1) {
           DFS(i)
           count++
       }
    }
    println(count)

}

fun DFS(i: Int) {
    visited[i] = 0
    matrix[i].forEach {
        if (visited[it] == 1) {
            visited[it] = 0
            DFS(it)
        }
    }

}

